### DISCUSSION TITLE
Multiple Presets or Views

### DISCUSSION THREAD

**Discussion by ghost** [directus/directus#9794] [Feature Requests]
Your probably more familiar with this concept if you work with or near Microsoft Dynamics 365 (which is a bit like an enterprise and very expensive version of Directus) but you can have multiple views on a table (a bit like presets) that a user can select from a dropdown. 

So you can have a preset with the Name "Items assigned to me" or "Unassigned Items" and depending on the preset you get different results defined by filters a bit like presets in directus.

The idea I am pitching is if a user creates multiple presets for a collection with the scope of all users a dropdown could appear allowing the user to select the preset they want to view. I've included a like explaining Dynamics Views below to hopefully make a bit more sense.

https://www.youtube.com/watch?v=zZ4g8-GoNdw&ab_channel=CRMDynamics

**Comment by benhaynes** [comment]
Interesting!! So, would this change mean opening a drop-down bookmark/preset menu when clicking some icon to the right of the collection name? I like it.

That would clean up the nav, and allow for grouping of bookmarks by: user, role, and global.

**Reply by amesas** [reply]
That would be first step. But Dynamics 365 allows not only basic grouping / filtering settings for the current view ( presets ).

Example of a more advanced setup: In excel with a dynamic table, changing that dynamic table settings, you  can display totally different info, from most sold products to average age per country for each of your store departments.

Same info displayed in a different way.

**Reply by benhaynes** [reply]
Sorry, I don't understand this response.

**Reply by amesas** [reply]
Updated. :)

**Reply by benhaynes** [reply]
Thanks! I believe we cover what you're mentioning already... but perhaps there are a few edge cases we don't have. Everything within the Collection view gets saved into the preset: search, filters, layout, layout options, sort field/order, items per page, etc. I'm not sure what else there is to save for different views... the only limitation is a preset is a window into a single/specific Collection (but that's by design, and relational data is still pulled in).

**Reply by benhaynes** [reply]
Just leaving this here for future reference... we might want to do something similar in the Nav too. Also, we could group based on User/Role/Global.

![Screen Shot 2021-11-16 at 9 39 04 AM](https://user-images.githubusercontent.com/522079/142005774-ff405702-ecb7-4eae-b333-1812c1db1e2b.png)

**Reply by ghost** [reply]
@benhaynes really like what you've done in the above picture separates the Global, Role, User with a line which looks really good. Only suggestion I might make it making it a bit clearer with a down angle icon or something next to the title showing that you can actually click on the title to get bookmarks as I feel it may be something people would only find if they accidentally clicked on the title?

**Reply by benhaynes** [reply]
Definitely, this was a pretty rough mock, and we'll clean it up before implementing. Thanks for the feedback!

**Reply by ghost** [reply]
@benhaynes I know your a busy so I won't take up too much of your time but any plan on implementing this sort of change in a future release?

**Reply by benhaynes** [reply]
No worries (but yes, very busy over here 🙂 ). Yes, we're planning to add this into the platform, however we've been getting a lot of "sponsored" work the past few weeks, and so this will likely be on the back-burner until it is sponsored, we get a PR to review (a good way to make this happen!), or eventually have idle time to build it internally.

Hope that helps!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
